guides/common/modules/con_puppet-architecture.adoc

```diff
@@ -0,0 +1,15 @@
+[id="puppet_guide_puppet_architecture_{context}"]
+= Puppet Architecture
+
+Puppet uses a server-client architecture with managed hosts running the Puppet agent and {Project} or any {SmartProxies} acting as a Puppet master.
+The Puppet agent enforces the declared state on managed hosts.
+It sends facts to the Puppet master, receives catalogs that are compiled from Puppet manifests, and sends reports to your {Project}.
+Differences between the declared state and the actual state are called _drifts_ in Puppet terminology.
+
+Puppet facts are information about the hosts running the Puppet agent.
+Running `facter` on a host displays available Puppet facts as _key-value_ pairs.
```

Comment:
Just running `facter` doesn't show everything. Modules can provide their own custom facts and Puppet adds a few. You can also run `puppet facts`. Running `facter` as an example is a good start, but we shouldn't give the impression it's all the facts there are.